src/voteHelper/VoteHelper.ts

```diff
@@ -0,0 +1,131 @@
+import { ApiPromise } from '@polkadot/api';
+import { SubstrateConnection } from '../connection/SubstrateConnection';
+import { NetworkConfig } from '../config/conf';
+import { sendTransaction } from '../utils';
+import { Referendum } from './referendum';
+import { AbstainVote, StandardVote, TrackVotes, Vote } from './votes';
+
+export class VoteHelper {
+    private api: ApiPromise;
+
+    constructor(connection: SubstrateConnection) {
+        this.api = connection.getApi();
+    }
+
+    async checkVotes(network: NetworkConfig, sender): Promise<void> {
+        const ongoingReferena = await this.getOngoingReferenda();
+        const potentialVotes = await this.getPotentialVotes(network, ongoingReferena);
+
+        for (const validator of network.validators) {
+            const newVotes = await this.decideHowToVote(validator, potentialVotes);
+
+            for (const vote of newVotes) {
+                await this.voteForReferenda(vote, sender, validator, network.voteAmount);
+            }
+        }
+    }
+
+    async getOngoingReferenda(): Promise<Referendum[]> {
+        const referendums = await this.api.query.referenda.referendumInfoFor.entries();
+        const castedReferendums = referendums.map(([key, value]) => new Referendum(key, value));
+        return castedReferendums.filter(referendum => referendum.data.ongoing);
+    }
+
+    async getAccountVotes(accountAddress: string): Promise<Vote[]> {
+        const uniqueVotes = new Set<number>();
+        const referendaTracks = this.api.consts.referenda.tracks;
+        const trackArray = (referendaTracks as any).map(([key, value]) => new TrackVotes(key, value));
+        const accountVotes: Vote[] = [];
+
+        for (const track of trackArray) {
+            const votesInTrack = await this.api.query.convictionVoting.votingFor(accountAddress, track.trackNumber);
+            (votesInTrack as any).asCasting.votes.map((voteData) => {
+                const vote = Vote.createVote(voteData);
+                if (!uniqueVotes.has(vote.referendaNumber)) {
+                    uniqueVotes.add(vote.referendaNumber);
+                    accountVotes.push(vote);
+                }
+            });
+        }
+
+        return accountVotes;
+    }
+
+    async getPotentialVotes(network: NetworkConfig, referendums: Referendum[]): Promise<Vote[]> {
+        const uniqueVotes = new Set<number>();
+        const newVoteForReferenda: Vote[] = [];
+        const abstainVotes: Vote[] = [];
+        const referendaNumbers = referendums.map(referendum => referendum.number);
```

Comment:
Speed optimization - converting `referendaNumbers` to Set will provide O(1) for `referendaNumbers.includes()`